File: aoj/introduction/IntroductiionToAlgorithmsAndDataStructures/1_1_A_Insertion_Sort/Main.java
import java.util.Scanner;

public class Main {

  public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int[] A = new int[N];

        for (int i = 0; i < N; i++) {
            A[i] = sc.nextInt();
        }

        insertionSort(A);

    }

    public static void insertionSort(int[] A) {

        for (int i = 0; i < A.length; i++) {

            int key = A[i];
            int j = i - 1;

            while (j >= 0 && A[j] > key) {

                A[j + 1] = A[j];
                j--;

            }

            A[j + 1] = key;
            print(A);

        }

    }

    public static void print(int[] A) {

        StringBuilder builder = new StringBuilder();

        for (int i = 0; i < A.length; i++) {
            builder.append(A[i] + " ");
        }

        System.out.println(builder.toString().trim());

    }

}

File: aoj/introduction/IntroductiionToAlgorithmsAndDataStructures/1_1_A_Insertion_Sort/Main.cpp
#include <iostream>

using namespace std;

void print(int A[], int n) {

    for (int i = 0; i < n; i++) {

        if (i != 0) {
            cout << ' ';
        }

        cout << A[i];

    }

    cout << endl;

}

void insertionSort(int A[], int n) {

    for (int i = 0; i < n; i++) {

        int key = A[i];
        int j = i - 1;

        while (j >= 0 && A[j] > key) {
            A[j + 1] = A[j];
            j--;
        }

        A[j + 1] = key;

        print(A, n);

    }

}

int main() {

    int N;
    cin >> N;

    int A[N];

    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }

    insertionSort(A, N);

    return 0;

}
